simple_model_a.py

```diff
@@ -0,0 +1,95 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+from sklearn.metrics import r2_score, mean_squared_error
+"""
+HERE U EXPLAIN WHAT THIS SCRIPT IS FOR. 
+"""
+
+
+def create_simple_ml_model(X_train, X_test, y_train, w_train, reg_):
+    """
+    helper function for make_simple_predictions_for_stacking(X_train, X_test, y_train, w_train, regressors)
+
+    :param X_train: X variables (floats) (training set)
+    :param X_test: X variables (floats) (testing set)
+    :param y_train: endogenous variable (float) (training set)
+    :param w_train: assignment to treatment (0 or 1) (training set)
+    :param reg_: a machine learning model instance
+    :return: the estimated treatment effect for the testing set,
+        as the difference between predicted outcome of treated and predicted outcome of untreated.
+    """
+
+    df = pd.DataFrame(data=X_train)
+    df['assignment'] = w_train
+
+    y_df = pd.DataFrame(data=y_train)
+    reg = reg_.fit(df, y_df)
+
+    X_neg = pd.DataFrame(data=X_test)
+    X_neg['assignment'] = 0
+
+    X_pos = pd.DataFrame(data=X_test)
+    X_pos['assignment'] = 1
+    ret = reg.predict(X_pos) - reg.predict(X_neg)
+    return ret
+
+
+def make_simple_predictions_for_stacking(X_train, X_test, y_train, w_train, regressors):
+    """
+    :param X_train: X variables (floats) (training set)
+    :param X_test: X variables (floats) (testing set)
+    :param y_train: endogenous variable (float) (training set)
+    :param w_train: assignment to treatment (0 or 1) (training set)
+    :param regressors: dictionary of name, regressor
+    :return: dict of regressors (name as string), predictions
+    """
+    predictions_dict = {}
+
+    for name in regressors.keys():
+        preds = create_simple_ml_model(X_train, X_test, y_train, w_train, regressors[name])
+        predictions_dict[name] = preds
+
+    return predictions_dict
+
+
+# TODO: add average line to plot, add true treatment effect to plot.
+def multilayer_hist(dictionary, true_vals):
+    """
+    :param dictionary: a dictionary of predictions, keys as name of model
+    :return: a plot of predicitons from each model in the dict
+    """
+
+    alpha = 0.2
+    bins = 250
+  
+    for name, predictions in dictionary.items():
+        plt.hist(predictions, alpha=alpha, bins=bins, label=name)
+        plt.hist(true_vals, alpha=alpha, bins=bins, label='te')
+
+        plt.axis([-5, 5, 0, 60])
```

Comment:
this from line 69 to 76 should not be moved into the for loop. you are plotting the axis, title, xlabel, ylabel, legend and SHOWING THE PLOT as many times as there are items in dictionary. this makes no sense.